ckb-vm-syscall-tracer/src/lib.rs

```diff
@@ -1018,6 +1018,98 @@ enum PartialLocator {
     Spawn(BinaryLocator),
 }
 
+#[derive(Default, Clone)]
+pub struct VmCreateCollector {
+    pending: Arc<Mutex<Option<(VmId, BinaryLocator)>>>,
```

Comment:
2 things:

* Personally, I would not simply track the creation of VM, for each spawn calls, I would recommend track the caller VM ID and the callee VM ID, the goal here is really to be able to build a tree structure of invocations of VMs within a single script group. Exec syscalls would also fit the tree structure.
* To me the invocation tree is in parallel to the BinaryLocator collector: there shall be one collector tracking the invocations between VMs, and there shall be another collector tracking the executed binary for each VM.